lib/TimespanLogging/src/core.jl

```diff
@@ -233,28 +233,47 @@ function write_event(ml::MultiEventLog, event::Event)
     end
 end
 
-function get_logs!(ml::MultiEventLog; only_local=false)
-    logs = Dict{Int,Dict{Symbol,Vector}}()
-    wkrs = only_local ? myid() : procs()
-    # FIXME: Log this logic
-    @sync for p in wkrs
-        @async begin
-            logs[p] = remotecall_fetch(p, ml) do ml
-                mls = get_state(ml)
-                lock(event_log_lock) do
-                    sublogs = Dict{Symbol,Vector}()
-                    for name in keys(mls.consumers)
-                        sublogs[name] = mls.consumer_logs[name]
-                        mls.consumer_logs[name] = []
+function get_logs!(ml::MultiEventLog; only_local=true)
+    if only_local
+        # Only return logs from the current process
+        logs = Dict{Int, Dict{Symbol,Vector}}()
+        pid = myid()
+        logs[pid] = remotecall_fetch(pid, ml) do ml 
```

Comment:
Why are you `remotecall_fetch`-ing here? The worker calling `get_logs!` with `only_local=true` is by definition the same worker that you need to fetch logs from, so you don't need that level of indirection.